packages/create/src/types/types.ts

```diff
@@ -0,0 +1,8 @@
+import type { Editor } from 'mem-fs-editor';
+
+/**
+ * mem-fs-editor types do not expose dump
+ */
+export type EditorWithDump = Editor & {
```

Comment:
While I don't remember when, but I defined that type before somewhere. I think we need it for testing all over the place, maybe it makes sense to introduce a `global.d.ts` in our repo and define that one there.

Reply:
We have a declaration in project access:
https://github.com/SAP/open-ux-tools/blob/46b8a2fe8db734da1b6f713d2bcca55310a59d77/packages/project-access/src/file/file-search.ts#L9-L21

How do you want to make this available cross modules?
https://www.typescriptlang.org/tsconfig#typeRoots
Or an include in the root `tsconfig.json`?

Reply:
I played with different ways and could not find a good solution. I also question now the approach we have in project-access. It works, but does redefine the `Editor` declaration, so other functions like `commit()` are missing. I think a better declaration would be:

```typescript
import type { Editor as MemFsEditor } from 'mem-fs-editor';
/**
 * Add missing dump function declaration as mem-fs-editor types do not expose this.
 */
declare module 'mem-fs-editor' {
    type FileMap = { [key: string]: { contents: string; state: 'modified' | 'deleted' } };

    export interface Editor extends MemFsEditor {
        /**
         * Dump files to compare expected result. Provide a cwd for relative path.
         * See also https://github.com/SBoudrias/mem-fs-editor#dumpcwd-filter for further details.
         *
         * @param [cwd] - optional, relative path
         */
        dump(cwd?: string): FileMap;
    }
}
```

But if we would have this in a central place in this monorepo, we to have a `devDependency` to `mem-fs-editor` on root level. 

II don't see a good and clean solution, ideas?

Reply:
How about doing it like https://github.com/SAP/open-ux-tools/pull/815? I branched from `main` and tried reusing it in `project-access` and in the tests of `fe-fpm-writer`.

Reply:
Why not simply update to https://github.com/SBoudrias/mem-fs-editor/releases/tag/v9.0.0 it now exposes `dump`?

Reply:
@IainSAP we are already using `v9.40` but the latest `@types/mem-fs-editor` module is `7.x` and is missing the `dump` function. So, we are not using a private function, just augment an incorrect type.
Or am I misunderstanding something?

Reply:
Hmm shouldn't we update the public types in that case and use any cast in the meantime, rather than managing types for third-party ourselves.

Reply:
Yes, check https://github.com/DefinitelyTyped/DefinitelyTyped#how-can-i-contribute. The first step is to do exactly what I did to make sure it works as expected. Once we have that proven, we contribute.

Reply:
> How about doing it like #815? I branched from `main` and tried reusing it in `project-access` and in the tests of `fe-fpm-writer`.

I like it, having a own private types module makes sense to me.

Reply:
Perfect, then I will get the PR ready for review asap.

Reply:
Changed to include `@sap-ux/types`.